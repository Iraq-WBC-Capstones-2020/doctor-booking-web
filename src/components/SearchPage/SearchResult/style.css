body {
  font-family: Roboto, sans-serif;
}

.SearchResult h3 {
  font-size: 18px;
  line-height: 24px;
  margin-bottom: 0px;
  color: #000000;
}

.SearchResult hr {
  margin-top: 0px;
  width: 100%;
}

.doctorCards {
  margin-top: 103px;
  width: 252px;
  background: #ffffff;
  -webkit-box-shadow: 0px 8px 24px rgba(40, 41, 61, 0.08);
  box-shadow: 0px 8px 24px rgba(40, 41, 61, 0.08);
  border-radius: 12px;
  border: none;
}

.doctorCards .card-title {
  margin-top: 50px;
}

.myImgCard {
  position: absolute;
  width: 72px !important;
  height: 64px !important;
  left: calc(50% - 72px / 2 + 4.5px);
  top: -58px;
  border-radius: 50%;
  -webkit-filter: drop-shadow(0px 4px 14px rgba(0, 0, 0, 0.15));
  filter: drop-shadow(0px 4px 14px rgba(0, 0, 0, 0.15));
}

.theButtons {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.theButtons button {
  height: 35px;
  width: 91px;
  font-weight: 500;
  font-size: 14px;
  line-height: 16px;
}

.theButtons .theWhite {
  color: #026fc2;
  background-color: white;
}

.theButtons .theBlue {
  color: #ffffff;
}

.specilty {
  font-size: 18px;
  line-height: 24px;
  margin-bottom: 29px;
  margin-top: 18px;
  color: #5d646a;
}
/*# sourceMappingURL=style.css.map */
