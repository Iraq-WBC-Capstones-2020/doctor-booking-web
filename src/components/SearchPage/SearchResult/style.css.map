{
    "version": 3,
    "mappings": "AAAA,AAAA,IAAI,CAAC;EACH,WAAW,EAAE,kBAAkB;CAChC;;AAED,AACE,aADW,CACX,EAAE,CAAC;EACD,SAAS,EAAE,IAAI;EACf,WAAW,EAAE,IAAI;EACjB,aAAa,EAAE,GAAG;EAClB,KAAK,EAAE,OAAO;CACf;;AANH,AAQE,aARW,CAQX,EAAE,CAAC;EACD,UAAU,EAAE,GAAG;EACf,KAAK,EAAE,IAAI;CACZ;;AAEH,AAAA,YAAY,CAAC;EACX,UAAU,EAAE,KAAK;EACjB,KAAK,EAAE,KAAK;EACZ,UAAU,EAAE,OAAO;EACnB,UAAU,EAAE,GAAG,CAAC,GAAG,CAAC,IAAI,CAAC,sBAAsB;EAC/C,aAAa,EAAE,IAAI;EACnB,MAAM,EAAE,IAAI;CAKb;;AAXD,AAQE,YARU,CAQV,WAAW,CAAC;EACV,UAAU,EAAE,IAAI;CACjB;;AAGH,AAAA,UAAU,CAAC;EACT,QAAQ,EAAE,QAAQ;EAClB,KAAK,EAAE,eAAe;EACtB,MAAM,EAAE,eAAe;EACvB,IAAI,EAAE,4BAA4B;EAClC,GAAG,EAAE,KAAK;EACV,aAAa,EAAE,GAAG;EAClB,MAAM,EAAE,6CAA6C;CACtD;;AAED,AAAA,WAAW,CAAC;EACV,OAAO,EAAE,IAAI;EACb,eAAe,EAAE,aAAa;CAmB/B;;AArBD,AAIE,WAJS,CAIT,MAAM,CAAC;EACL,MAAM,EAAE,IAAI;EACZ,KAAK,EAAE,IAAI;EACX,WAAW,EAAE,GAAG;EAChB,SAAS,EAAE,IAAI;EACf,WAAW,EAAE,IAAI;CAClB;;AAVH,AAYE,WAZS,CAYT,SAAS,CAAC;EACR,KAAK,EAAE,OAAO;EACd,gBAAgB,EAAE,KAAK;CACxB;;AAfH,AAiBE,WAjBS,CAiBT,QAAQ,CAAC;EACP,KAAK,EAAE,OAAO;CACf;;AAIH,AAAA,SAAS,CAAC;EACR,SAAS,EAAE,IAAI;EACf,WAAW,EAAE,IAAI;EACjB,aAAa,EAAE,IAAI;EACnB,UAAU,EAAE,IAAI;EAChB,KAAK,EAAE,OAAO;CACf;;AAED,AAAA,SAAS,CAAA;EACP,UAAU,EAAE,cAAc;CAC3B",
    "sources": [
        "style.scss"
    ],
    "names": [],
    "file": "style.css"
}