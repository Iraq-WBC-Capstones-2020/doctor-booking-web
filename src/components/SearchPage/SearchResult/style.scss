.SearchResult{
    h3{
        font-family: Roboto;
        font-style: normal;
        font-weight: 500;
        font-size: 18px;
        line-height: 24px;
        text-align: left;
        margin-bottom: 0px;
        color: #000000;

    }
    hr{
        margin-top: 0px;
        width: 100%;
    }
}
.doctorCards{
    margin-top: 103px;
    background: #FFFFFF;
box-shadow: 0px 8px 24px rgba(40, 41, 61, 0.08);
border-radius: 12px;
border: none;
}
.card-img-top{
    
    position: absolute;
width: 72px;
height: 64px;
left: calc(50% - 72px/2 + 4.5px);
top: -58px;
border-radius: 50%;
filter: drop-shadow(0px 4px 14px rgba(0, 0, 0, 0.15));


}
.theButtons{
    display: flex;
    justify-content: space-between;
    button{
        height: 35px ;
        width:91px ;
        font-family: Roboto;
        font-style: normal;
        font-weight: 500;
        font-size: 14px;
        line-height: 16px;
        /* identical to box height */

    }
    .theWhite{
      

        text-align: center;

        color: #026FC2;
        background-color: white;
    }

    .theBlue{
        color: #FFFFFF;
    }
}

.specilty{
    font-family: Roboto;
    font-style: normal;
    font-weight: normal;
    font-size: 18px;
    line-height: 24px;
    align-items: center;
    text-align: center;
    margin-bottom: 29px;
    margin-top: 18px;

    color: #5D646A;

}

.card-text{
    font-family: Roboto;
    font-style: normal;
    font-weight: normal;
    font-size: 18px;
    line-height: 24px;
    /* or 133% */

    display: flex;
    align-items: center;
    text-align: center;

    color: #5D646A;
}