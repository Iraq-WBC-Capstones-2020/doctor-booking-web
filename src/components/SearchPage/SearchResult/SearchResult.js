import React, { useContext } from 'react';
import { Container, Row, Col, Button, Card } from 'react-bootstrap';
import Avatar from './images/avatar.png';
import './style.css';
import { DoctorContext } from '../../../DoctorContext';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';

function SearchResult({ filterDoctors }) {
  const [state] = useContext(DoctorContext);

  SearchResult.propTypes = {
    filterDoctors: PropTypes.array,
  };

  return (
    <div className="searchResult">
      <Container>
        <Row>
          <Col>
            {filterDoctors?.length === 0 ? (
              <h2 className="text-primary">Doctor List</h2>
            ) : (
              <h2 className="text-success">Search result</h2>
            )}
          </Col>
          <hr className="solid" />
        </Row>

        {filterDoctors?.length === 0 ? (
          <Row className="cardsRow">
            {state.doctors.map((doctor, index) => (
              <Col md={4} key={index}>
                <Card className="doctorCards " style={{ width: '18rem' }}>
                  <Card.Img className="myImgCard" variant="top" src={Avatar} />
                  <Card.Body>
                    <Card.Title>{doctor.name}</Card.Title>
                    <div className="specilty">{doctor.speciality}</div>
                    <Card.Text>
                      {' '}
                      Lorem, ipsum dolor amet consectetur adipisicing elit.
                      Provident, totam?{' '}
                    </Card.Text>
                    <div className="theButtons">
                      <Link to={`profile/${doctor.uid}`}>
                        <Button className="theWhite" variant="primary">
                          More Info
                        </Button>
                      </Link>
                      <Link to={`booking/${doctor.uid}`}>
                        <Button className="theBlue" variant="primary">
                          Book now
                        </Button>
                      </Link>
                    </div>
                  </Card.Body>
                </Card>
              </Col>
            ))}
          </Row>
        ) : (
          <Row className="cardsRow">
            {filterDoctors?.map((doctor, index) => (
              <Col md={4} key={index}>
                <Card className="doctorCards " style={{ width: '18rem' }}>
                  <Card.Img className="myImgCard" variant="top" src={Avatar} />
                  <Card.Body>
                    <Card.Title>{doctor.name}</Card.Title>
                    <div className="specilty">{doctor.speciality}</div>
                    <Card.Text>
                      {' '}
                      Lorem, ipsum dolor amet consectetur adipisicing elit.
                      Provident, totam?{' '}
                    </Card.Text>
                    <div className="theButtons">
                      <Link to={`profile/${doctor.uid}`}>
                        <Button className="theWhite" variant="primary">
                          More Info
                        </Button>
                      </Link>
                      <Link to={`booking/${doctor.uid}`}>
                        <Button className="theBlue" variant="primary">
                          Book now
                        </Button>
                      </Link>
                    </div>
                  </Card.Body>
                </Card>
              </Col>
            ))}
          </Row>
        )}
      </Container>
    </div>
  );
}

export default SearchResult;
