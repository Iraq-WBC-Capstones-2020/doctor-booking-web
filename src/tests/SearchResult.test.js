import React from 'react';
import { render } from '@testing-library/react';
import SearchResult from '../components/SearchPage/SearchResult/SearchResult';
import renderer from 'react-test-renderer';
import DoctorInfoProvider from '../DoctorContext';
it('render1', () => {
  render(
    <DoctorInfoProvider>
      <SearchResult length={0} />
    </DoctorInfoProvider>
  );
});

it('renders correctly', () => {
  const tree = renderer
    .create(
      <DoctorInfoProvider>
        <SearchResult length={0} />
      </DoctorInfoProvider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
