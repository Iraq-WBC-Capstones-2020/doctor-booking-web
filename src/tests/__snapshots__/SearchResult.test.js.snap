// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="searchResult"
>
  <div
    className="container"
  >
    <div
      className="row"
    >
      <div
        className="col"
      >
        <h3>
          Search result
        </h3>
      </div>
      <hr
        className="solid"
      />
    </div>
    <div
      className="cardsRow row"
    >
      <div
        className="col-md-4"
      >
        <div
          className="doctorCards card"
          style={
            Object {
              "width": "18rem",
            }
          }
        >
          <img
            className="card-img-top myImgCard"
            src="avatar.png"
          />
          <div
            className="card-body"
          >
            <div
              className="card-title h5"
            >
              Zainab Mohammed
            </div>
            <div
              className="specilty"
            >
              Emergency Medicine
            </div>
            <p
              className="card-text"
            >
              Some quick example text to build on the card title and make up the bulk of the card's content.
            </p>
            <div
              className="theButtons"
            >
              <button
                className="theWhite btn btn-primary"
                disabled={false}
                type="button"
              >
                More Info
              </button>
              <button
                className="theBlue btn btn-primary"
                disabled={false}
                type="button"
              >
                Book now
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        className="col-md-4"
      >
        <div
          className="doctorCards card"
          style={
            Object {
              "width": "18rem",
            }
          }
        >
          <img
            className="card-img-top myImgCard"
            src="avatar.png"
          />
          <div
            className="card-body"
          >
            <div
              className="card-title h5"
            >
              Ali Fareed
            </div>
            <div
              className="specilty"
            >
              Child Doc
            </div>
            <p
              className="card-text"
            >
              Some quick example text to build on the card title and make up the bulk of the card's content.
            </p>
            <div
              className="theButtons"
            >
              <button
                className="theWhite btn btn-primary"
                disabled={false}
                type="button"
              >
                More Info
              </button>
              <button
                className="theBlue btn btn-primary"
                disabled={false}
                type="button"
              >
                Book now
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        className="col-md-4"
      >
        <div
          className="doctorCards card"
          style={
            Object {
              "width": "18rem",
            }
          }
        >
          <img
            className="card-img-top myImgCard"
            src="avatar.png"
          />
          <div
            className="card-body"
          >
            <div
              className="card-title h5"
            >
              Sabah Omar
            </div>
            <div
              className="specilty"
            >
              Family Medicine
            </div>
            <p
              className="card-text"
            >
              Some quick example text to build on the card title and make up the bulk of the card's content.
            </p>
            <div
              className="theButtons"
            >
              <button
                className="theWhite btn btn-primary"
                disabled={false}
                type="button"
              >
                More Info
              </button>
              <button
                className="theBlue btn btn-primary"
                disabled={false}
                type="button"
              >
                Book now
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        className="col-md-4"
      >
        <div
          className="doctorCards card"
          style={
            Object {
              "width": "18rem",
            }
          }
        >
          <img
            className="card-img-top myImgCard"
            src="avatar.png"
          />
          <div
            className="card-body"
          >
            <div
              className="card-title h5"
            >
              Sarah Marwan
            </div>
            <div
              className="specilty"
            >
              Child Doc
            </div>
            <p
              className="card-text"
            >
              Some quick example text to build on the card title and make up the bulk of the card's content.
            </p>
            <div
              className="theButtons"
            >
              <button
                className="theWhite btn btn-primary"
                disabled={false}
                type="button"
              >
                More Info
              </button>
              <button
                className="theBlue btn btn-primary"
                disabled={false}
                type="button"
              >
                Book now
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        className="col-md-4"
      >
        <div
          className="doctorCards card"
          style={
            Object {
              "width": "18rem",
            }
          }
        >
          <img
            className="card-img-top myImgCard"
            src="avatar.png"
          />
          <div
            className="card-body"
          >
            <div
              className="card-title h5"
            >
              Salih Murtadha
            </div>
            <div
              className="specilty"
            >
              Medical Genetics
            </div>
            <p
              className="card-text"
            >
              Some quick example text to build on the card title and make up the bulk of the card's content.
            </p>
            <div
              className="theButtons"
            >
              <button
                className="theWhite btn btn-primary"
                disabled={false}
                type="button"
              >
                More Info
              </button>
              <button
                className="theBlue btn btn-primary"
                disabled={false}
                type="button"
              >
                Book now
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        className="col-md-4"
      >
        <div
          className="doctorCards card"
          style={
            Object {
              "width": "18rem",
            }
          }
        >
          <img
            className="card-img-top myImgCard"
            src="avatar.png"
          />
          <div
            className="card-body"
          >
            <div
              className="card-title h5"
            >
              Shadya Naji
            </div>
            <div
              className="specilty"
            >
              Child Doc
            </div>
            <p
              className="card-text"
            >
              Some quick example text to build on the card title and make up the bulk of the card's content.
            </p>
            <div
              className="theButtons"
            >
              <button
                className="theWhite btn btn-primary"
                disabled={false}
                type="button"
              >
                More Info
              </button>
              <button
                className="theBlue btn btn-primary"
                disabled={false}
                type="button"
              >
                Book now
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
